src/Integration/State/StateManager.cs

```diff
@@ -176,6 +196,14 @@ private void SetProjectsUIThread(ConnectionInformation connection, IEnumerable<P
             }
         }
 
+        private void DisposeConnections()
+        {
+            this.ManagedState.ConnectedServers
```

Comment:
If `TransferableVisualState` holds disposable objects then shouldn't it also be disposable?
And the transitively this `StateManager` can call `DisposeConnections` in the dispose method.